cpp/arcticdb/async/tasks.cpp

```diff
@@ -36,8 +36,8 @@ namespace arcticdb::async {
     }
 
     pipelines::SegmentAndSlice DecodeSliceTask::decode_into_slice(storage::KeySegmentPair&& key_segment_pair) {
-        auto key = std::move(key_segment_pair.atom_key());
-        auto seg = std::move(key_segment_pair.release_segment());
+        auto key = key_segment_pair.atom_key();
+        auto& seg = key_segment_pair.segment();
```

Comment:
Can this be `const&`?

Reply:
No, because of `decode_into_memory_segment`. Sorting that out forces us in to the "make Segment logically const" refactor that we declined.

Reply:
What happens in decode_into_memory_segment?